app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,32 @@
+class GroupsController < ApplicationController
+
+  def new
+    @group = Group.new
+  end
+
+  def create
+    @group = Group.create(group_params)
+    redirect_to groups_url
+  end
+
+  def index
+    @groups = Group.all
+  end
+
+  def show
+    @group = Group.find(params[:id])
+    @users = @group.users
+  end
+
+  def update
```

Comment:
This is not updating a Group, it is creating a Membership. Also, it is non-idempotent, meaning you could call this action twice and you would get 2 similar Membership rows. PUT http requests should be idempotent. http://restcookbook.com/HTTP%20Methods/put-vs-post/

Not a blocker, but feels un-railsy - you could consider a refactor.